stl/inc/xlocnum

```diff
@@ -1312,18 +1317,23 @@ protected:
         _OutIt _Dest, ios_base& _Iosbase, _Elem _Fill, long double _Val) const { // put formatted long double to _Dest
         string _Buf;
         char _Fmt[8];
-        bool _Isfixed         = (_Iosbase.flags() & ios_base::floatfield) == ios_base::fixed;
-        streamsize _Precision = _Iosbase.precision() <= 0 && !_Isfixed ? 6 : _Iosbase.precision(); // desired precision
-        size_t _Bufsize       = static_cast<size_t>(_Precision);
+        const bool _Isfixed = (_Iosbase.flags() & ios_base::floatfield) == ios_base::fixed;
+        const bool _Ishex   = (_Iosbase.flags() & ios_base::floatfield) == (ios_base::fixed | ios_base::scientific);
+        const streamsize _Precision =

```

Comment:
again, the ternary here is getting a little complicated for my liking, but again, I get that ti's a const variable.